libs/test-common/src/main/java/org/codice/ddf/test/common/mockito/PrivilegedVerificationMode.java

```diff
@@ -0,0 +1,142 @@
+import org.mockito.internal.verification.api.VerificationInOrderMode;
+import org.mockito.invocation.Invocation;
+import org.mockito.invocation.Location;
+import org.mockito.invocation.MatchableInvocation;
+import org.mockito.verification.VerificationMode;
+
+/**
+ * Class that extends Mockito's {@link VerificationMode} to verify that a mocked method has been
+ * called within a {@code AccessController#doPrivileged()} block.
+ *
+ * <p><b> This code is experimental. While this class is functional and tested, it may change or be
+ * removed in a future version of the library. </b>
+ */
+@Beta
+public class PrivilegedVerificationMode implements VerificationMode, VerificationInOrderMode {
+
+  private final VerificationMode mode;
+
+  private PrivilegedVerificationMode(VerificationMode mode) {
+    this.mode = mode;
+  }
+
+  /**
+   * Wraps another {@link VerificationMode} instance and ensures all the expected calls were made
+   * inside a {@code AccessController.doPrivileged()} block.
+   *
+   * @param mode {@link VerificationMode} instance to wrap with a privileged check
+   * @return wrapped {@link VerificationMode} instance
+   */
+  public static VerificationMode privileged(VerificationMode mode) {
+    return new PrivilegedVerificationMode(mode);
+  }
+
+  /**
+   * Returns a {@link VerificationMode} object that ensures a single expected calls was made inside
+   * a {@code AccessController.doPrivileged()} block.
+   *
+   * @return wrapped {@link VerificationMode} instance
+   */
+  public static VerificationMode privileged() {
+    return new PrivilegedVerificationMode(times(1));
+  }
+
+  @Override
+  public void verify(VerificationData verificationData) {
+    String target = verificationData.getTarget().toString();
+
+    mode.verify(verificationData);
+
+    verify(verificationData.getTarget(), verificationData.getAllInvocations());
+  }
+
+  @Override
+  public void verifyInOrder(VerificationDataInOrder verificationDataInOrder) {
+    String target = verificationDataInOrder.getWanted().toString();
+
+    if (!(mode instanceof VerificationInOrderMode)) {
+      throw new MockitoAssertionError(mode.toString() + " cannot be used in ordered mode");
+    }
+
+    ((VerificationInOrderMode) mode).verifyInOrder(verificationDataInOrder);
+
+    verify(verificationDataInOrder.getWanted(), verificationDataInOrder.getAllInvocations());
+  }
+
+  @Override
+  public VerificationMode description(String description) {
+    return VerificationModeFactory.description(this, description);
+  }
+
+  private void verify(MatchableInvocation wanted, List<Invocation> invocations) {
+
+    if (invocations.size() == 0) {
+      return;
+    }
+
+    try {
+      Location location = invocations.get(0).getLocation();
+      Field stackTraceHolderField = getStackTraceHolderField(location);
+
+      for (Invocation invocation : invocations) {
+        if (wanted.matches(invocation)) {
+          if (getStackTraceElements(stackTraceHolderField, invocation)
```

Comment:
❓ Do we know at this stage when was their `stackTraceHolder`? I am asking because if we knew it was taken when the actual calls was made, we could just go back to the previous entry or such instead of going back in times so many frames. In addition, we could take our own holder (a.k.a. a throwable) inside the `verify()` methods which could help filter off stack frames that are there in both; that is the stack frames from whatever called the test method in the first place.